CMakeLists.txt

```diff
@@ -38,6 +38,7 @@ option(IREE_BUILD_BENCHMARKS "Builds IREE benchmark suites." OFF)
 option(IREE_BUILD_DOCS "Builds IREE docs." OFF)
 option(IREE_BUILD_SAMPLES "Builds IREE sample projects." ON)
 option(IREE_BUILD_EMBEDDING_SAMPLES "Builds IREE embedding sample projects. The compiler needs to be available." OFF)
+option(IREE_BUILD_STATIC_LIBRARY_SAMPLES "Builds IREE static library sample projects. Requires compiling static libraries before running samples." OFF)
```

Comment:
we should be able to key this off of other cmake variables - a good way to think about this is that every cmake variable we add is a new CI bot we should have - and note today that we have dozens and only a handful of bots ;)

@ScottTodd may know what the current set we use to see if iree-translate is available (IREE_BUILD_COMPILER is if we build it, but I think there's one that works if the compiler is available on the path) - that and IREE_BUILD_SAMPLES should be the only thing gating this

Reply:
I will add @benvanik this sample won't run without user action. Scott and I have discussed how we could automate it using scripts to run the static output from the compiler first, but as it is now, if you enable this flag, its a broken target. 

If we combine this with other flags, I'm not certain the effects of the broken target will be to continuous builds. 

I totally don't disagree we have a lot flags. If there is a significant cost to adding each one, perhaps we should consolidate demos under 1 or 2 flags instead of one for each?

Reply:
I can take a closer look at this in a bit, but here's an idea:

Check for the existence of the file that needs to be built manually, rather than adding a top level option.

---

From another angle: `iree_bytecode_module` may work, but I don't want users depending on our CMake helper functions, so I don't want to use it in samples.

iree-template-cpp has some "raw" CMake that does roughly what I'd want us to use here: https://github.com/iml130/iree-template-cpp/blob/92f8b06f2c6956e37397ff835796fdbbcb84cb7d/simple_embedding/CMakeLists.txt#L13-L57

If that approach works and the generated .o file can be depended on naturally, then we wouldn't need to check for a special option or the existence of a manually compiled file.

Reply:
We can either put the `iree-translate` -> `generate_embed_data` in CMake like iree-template-cpp, or put it in a list of manual steps (and an automated test.sh or build.sh script). I'm sort of leaning towards the manual steps + automated verification for samples, but open to other perspectives :)

Reply:
Here you go: https://github.com/KoolJBlack/iree/pull/1 (feel free to adapt the changes there or merge and iterate on this PR)
* works without manual steps or checked in binary files
* tested on Windows


I'm still on the fence regarding raw CMake vs using our helpers.

Reply:
Thanks @ScottTodd. I've incorporated this into the demo.

Reply:
You can remove the `IREE_BUILD_STATIC_LIBRARY_SAMPLES` option now